### ISSUE TITLE
Question about k anonymity metric

### ISSUE THREAD

**Issue by amad-person** [vanderschaarlab/synthcity#273]
I had a question about interpreting [synthcity's k anonymity metric](https://github.com/vanderschaarlab/synthcity/blob/943fa280687d236d783e53f40302838f5924f422/src/synthcity/metrics/eval_privacy.py#L70) for a synthetic dataset. 

Consider the following example train dataset:

|     | Age | Gender | Zip Code | Medical Condition | 
|-----|-----|--------|----------|-------------------|
| 1   | 25  | F      | 10000    | Condition X       |
| ... | ... | ...    | ...      | ...               |
| n   | 30  | M      | 20000    | Condition Y       | 

Here, the sensitive feature is `Medical Condition`. 

Suppose a synthetic dataset has `k = 1` because there is only one such row in it:

|     | Age | Gender | Zip Code | Medical Condition | 
|-----|-----|--------|----------|-------------------|
| 1   | 25  | F      | 10000    | Condition Y       |

Here, the sensitive feature (`Condition Y`) in the synthetic dataset is not the true one in the train dataset (`Condition X`). So on observing the synthetic dataset, the adversary won't learn the true value for the sensitive feature. In this case, can we say that a low k value for the synthetic dataset doesn't necessarily imply it has lesser privacy? 

Are there any recommended guidelines on interpreting synthcity's k anonymity metric? 

Thank you!

**Comment by robsdavis** [comment]
Hi @amad-person, Thanks for your important question!

You are correct, in the exact example you outline, the k may be one and yet the privacy concerns may not necessarily be as severe as in other cases. However, in high-fidelity, synthetic datasets there may be many records that are close to real records and have the same value for sensitive features. A high k value in this instance is more important than in your example. 

A low k value in synthetic data does not inherently indicate a severe privacy concern, especially when the sensitive attributes do not match those in the original dataset. However, a higher k value will always be a good indicator that the data is likely to have fewer privacy concerns.

So, in summary, k-anonimity is not a perfect metric that captures all nuances of the privacy of a dataset, but can be used in concert with other metrics to better understand the level of privacy in the data.
